axelor-account/src/main/java/com/axelor/apps/account/service/IrrecoverableService.java

```diff
@@ -872,102 +875,87 @@ public Move createIrrecoverableMove(
     } else if (invoice.getMove().getOrigin() != null && irrecoverableName == null) {
       originStr = invoice.getMove().getOrigin();
     }
+
     if (isInvoiceReject) {
       creditAmount = invoice.getRejectMoveLine().getAmountRemaining();
-      debitAmount = creditAmount;
     } else {
       creditAmount = invoice.getCompanyInTaxTotalRemaining();
-      debitAmount = creditAmount;
     }
 
     // Debits MoveLines Tva
-    for (InvoiceLineTax invoiceLineTax : invoice.getInvoiceLineTaxList()) {
+    for (MoveLine invoiceMoveLine : invoiceMove.getMoveLineList()) {
+
+      if (invoiceMoveLine.getDebit().compareTo(BigDecimal.ZERO) > 0) {
+        amount = invoiceMoveLine.getAmountRemaining();
+
+        // Credit MoveLine Customer account (411, 416, ...)
+        MoveLine creditMoveLine =
+            moveLineCreateService.createMoveLine(
+                move,
+                payerPartner,
+                invoiceMoveLine.getAccount(),
+                amount,
+                false,
+                appAccountService.getTodayDate(company),
+                seq,
+                originStr,
+                invoice.getInvoiceId());
+        move.getMoveLineList().add(creditMoveLine);
+
+        Reconcile reconcile =
+            reconcileService.createReconcile(invoiceMoveLine, creditMoveLine, amount, false);
+        if (reconcile != null) {
+          reconcileService.confirmReconcile(reconcile, true);
+        }
```

Comment:
It should not be done here, the InvoiceTerms have not been created on the creditMoveLine yet. 
Instead, it should be done after line 542, when we call moveValidateService.accounting(move);